Simplify XML parsing code

AliSoftware opened the pull request:
Migrate all AWSResponses XML parsing from SAX (`XMLParser` + `delegate`) to DOM (`XMLDocument`)

This is an approach that I've started to use when I [implemented additional AWS requests for resumable upload support](https://github.com/Automattic/hostmgr/pull/89), and it simplifies the parsing code drastically, so I figured I'd migrate existing parsers to use this approach too.

---
_Note: This builds on top of https://github.com/Automattic/hostmgr/pull/90 / `refine-tests` branch—so that we benefit from tests not referencing internal Parser implementations (that used a SAX `XMLParser`) anymore._

AliSoftware committed:
Migrate all XML parsing to DOM (XMLDocument)

To make the code simpler, instead of using XMLParser's SAX / event-driven approach

AliSoftware committed:
Improve/Simplify XML parsing code even more

 - Add `AWSResponse.stub` to `TestHelpers.swift`
 - Add `XMLDocument.rootElement(expectedName:)` and `XMLElement.text(forElementName:)` to `Helpers.swift` and make the parsing code simpler

This also makes the lines shorter, helping making `swiftlint` happy.

crazytonyli commented on Sources/tinys3/Helpers.swift:
```diff
@@ -1,24 +1,37 @@
-            return
+extension XMLElement {
+    func value<T>(forElementName name: String, transform: (String) -> T? = { $0 }) throws -> T {
+        guard let text = self.elements(forName: name).first?.stringValue else {
```
Question: do you think this function should throw an error if there are more than one element for the given name?

AliSoftware replied:
Good idea. Added in 82c9d038c01756cfd40e59ea2397c8923515f28e

AliSoftware committed:
Throw error if multiple elements of expected name

As suggested in https://github.com/Automattic/hostmgr/pull/91#discussion_r1536930684
